### PR TITLE
QUIC: NetEvent and UnixNetVConnection

### PR DESCRIPTION
**Pull request by scw00**

### CODE REVIEW

**Comment by scw00** [comment]
Introduce new class `NetEventHandler ` to Split `NetHandler` and `UnixNetVConnection`.

In this current implementation. The `NetHandler` is bond to `UnixNetVConnection`. So  if someone wants to use `NetHandler` to implement their own nets events (like udp), it should derived from `UnixNetVConnection` that will make the object too large. 

So abstract necessary api to hide more detail of implementation.

**Comment by scw00** [comment]
[skip ci]

**Comment by scw00** [comment]
[approve ci autest]

**Comment by scw00** [comment]
image

When udp packet coming, `AccpetUDP2Connection` will  create a new `UDP2Connection` with new `fd` to accept matched five tuple UDPPacket. And Then late UDPPacket with matched five tuple will be accepted by the new established `UDP2Connection`.

File: iocore/net/NetEvent.h [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  virtual Ptr<ProxyMutex> &get_mutex()      = 0;
+  virtual ContFlags &get_control_flags()    = 0;
+  virtual sockaddr const *get_remote_addr() = 0;
+  virtual const NetVCOptions &get_options() = 0;
+
+  EventIO ep{};
+  NetState read{};
+  NetState write{};
+
+  bool closed    = false;
```
Comment (oknet) [inline iocore/net/NetEvent.h]: `int closed  = 0` ?

Comment (scw00) [reply iocore/net/NetEvent.h]: fixed

**Comment by scw00** [comment]
close it and open a new pr #6316

**Pull request closed without merging**